lib/ansible/modules/network/netconf/netconf_get.py

```diff
@@ -0,0 +1,234 @@
+  netconf_get:
+    display: json
+
+- name: get schema list using subtree w/ namespaces
+  netconf_get:
+    format: json
+    filter: <netconf-state xmlns="urn:ietf:params:xml:ns:yang:ietf-netconf-monitoring"><schemas><schema/></schemas></netconf-state>
+    lock: False
+
+- name: get schema list using xpath
+  netconf_get:
+    format: json
+    filter: /netconf-state/schemas/schema
+
+- name: get interface confiugration with filter (iosxr)
+  netconf_get:
+    filter: <interface-configurations xmlns="http://cisco.com/ns/yang/Cisco-IOS-XR-ifmgr-cfg"></interface-configurations>
+"""
+
+RETURN = """
+stdout:
+  description: The raw XML string containing configuration or state data
+               received from the underlying ncclient library.
+  returned: always apart from low-level errors (such as action plugin)
+  type: string
+  sample: '...'
+stdout_lines:
+  description: The value of stdout split into a list
+  returned: always apart from low-level errors (such as action plugin)
+  type: list
+  sample: ['...', '...']
+output:
+  description: Based on the value of display option will return either the set of
+               transformed XML to JSON format from the RPC response with type dict
+               or pretty XML string response (human-readable) or response with
+               namespace removed from XML string.
+  returned: when the display format is selected as JSON it is returned as dict type, if the
+            display format is xml or pretty pretty it is retured as a string apart from low-level
+            errors (such as action plugin).
+  type: complex
+  contains:
+    formatted_output:
+      - Contains formatted response received from remote host as per the value in display format.
+"""
+import sys
+
+try:
+    from lxml.etree import Element, SubElement, tostring, fromstring, XMLSyntaxError
+except ImportError:
+    from xml.etree.ElementTree import Element, SubElement, tostring, fromstring
+    if sys.version_info < (2, 7):
+        from xml.parsers.expat import ExpatError as XMLSyntaxError
+    else:
+        from xml.etree.ElementTree import ParseError as XMLSyntaxError
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.network.netconf.netconf import get_capabilities, locked_config, get_config, get
+from ansible.module_utils.network.common.netconf import remove_namespaces
+
+try:
+    import jxmlease
+    HAS_JXMLEASE = True
+except ImportError:
+    HAS_JXMLEASE = False
+
+
+def get_filter_type(filter):
+    if not filter:
+        return None
+    else:
+        try:
+            fromstring(filter)
+            return 'subtree'
+        except XMLSyntaxError:
+            return 'xpath'
+
+
+def main():
+    """entry point for module execution
+    """
+    argument_spec = dict(
+        source=dict(choices=['running', 'candidate', 'startup']),
+        filter=dict(),
+        display=dict(choices=['json', 'pretty', 'xml']),
+        lock=dict(default=False, type=bool)
```

Comment:
Change this argument from bool to choices with the following choices `[never, always, if-supported]`

In the `never` case, do not attempt to lock the conf data store
In the 'always` case, try to lock the conf data store and error if the lock operation failes
In the 'if-supported' case, try to lock the conf data store and continue unlocked if the lock operation fails

The default value should be `never`